src/main/java/org/apache/sling/feature/launcher/impl/Main.java

```diff
@@ -174,8 +179,9 @@ protected static void parseArgs(final LauncherConfig config, final String[] args
         final Option fwkProperties = Option.builder(OPT_FRAMEWORK_PROPERTIES)
                 .longOpt("framework-properties")
                 .desc("Set framework properties")
+                .hasArg()
                 .optionalArg(true)
```

Comment:
I don't believe it makes sense for the arg to be optional here, so maybe remove this line?

Reply:
The optional arg is important for the docker setup. In the docker file we delegate every env to a option. So it could be that most Options are empty. And if we do jott have opt arg the cliparser will fail.

Reply:
I see.  Would you be opposed to changing the Dockerfile to include the "-D" in the FRAMEWORK_PROPERTIES env value string before each name=value and separate each of those items with a space?  That might make it simpler to parse and avoid troubles with escaping the commas (if any) in the property name or value.

Reply:
will fix this also here

Reply:
Injecting the "-D " like that might work, but I was initially thinking to just document the syntax and expect the user to supply the "-D " segments manually with something like this:
```
# NOTE: expected syntax: -D name1=value1 -D name2=value2
FRAMEWORK_PROPERTIES="-D a=1 -D b=2 -D c=3"
```
I'm not sure the additional complexity is necessary.

Reply:
I see, will reduce it

Reply:
It looks good to me.